### PR TITLE
Add WebSocket & API Connection

### PR DESCRIPTION
**Pull request by furiner**
Just an active and constant connection to a HTTP server, and a WebSocket; namely for active communications with the server and keeping up with replicated and consistent data, such as session tokens.

### CODE REVIEW

File: Kumi.Game/Online/API/Accounts/APIAccount.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+﻿using Newtonsoft.Json;
+
+namespace Kumi.Game.Online.API.Users;
+
+public class APIAccount
+{
+    /// <summary>
+    /// A special ID used to represent the system users, or any other user that is not logged in.
+    /// </summary>
+    public const int DEFAULT_SYSTEM_ID = 0;
```
Comment (Ryooshuu) [inline Kumi.Game/Online/API/Accounts/APIAccount.cs:10]: why not -1?

**Comment by furiner** [comment]
Had to manually add because PR size bot was added after this pull request's creation.

File: Kumi.Game/Online/API/DummyAPIConnection.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    public void Login(string username, string password)
+    {
+        State.Value = APIState.Connecting;
+
+        // Create a new account object.
+        LocalAccount.Value = new APIAccount
+        {
+            Id = 1,
+            Username = "Dummy",
+        };
```
Comment (Ryooshuu) [inline Kumi.Game/Online/API/DummyAPIConnection.cs:69]: May want to use `GuestAccount` instead...

File: Kumi.Game/Online/API/IAPIConnectionProvider.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+using Kumi.Game.Online.API.Users;
+using Kumi.Game.Online.Server;
+using osu.Framework.Bindables;
+
+namespace Kumi.Game.Online.API;
+
+/// <summary>
+/// An object that represents a connection to the API.
+/// </summary>
+public interface IAPIConnectionProvider
```
Comment (Ryooshuu) [inline Kumi.Game/Online/API/IAPIConnectionProvider.cs:11]: Where's the actual implementation for this interface that's not the dummy API connection?

**Review by Ryooshuu — CHANGES_REQUESTED** [review]

File: Kumi.Game/Online/API/Accounts/APIAccount.cs (hunk 1 above)
Comment (furiner) [reply Kumi.Game/Online/API/Accounts/APIAccount.cs:10]: IDs are generally unsigned, and there'll never be an account with the ID of 0.

File: Kumi.Game/Online/API/DummyAPIConnection.cs (hunk 2 above)
Comment (furiner) [reply Kumi.Game/Online/API/DummyAPIConnection.cs:69]: This is just a fake account upon login, to show that it isn't just the default guest account used only when not logged in.

File: Kumi.Game/Online/API/IAPIConnectionProvider.cs (hunk 3 above)
Comment (furiner) [reply Kumi.Game/Online/API/IAPIConnectionProvider.cs:11]: Waiting for whenever I push to [kikoyudev/kumi-api](https://github.com/kikuyodev/kumi-api).

**Review by Ryooshuu — APPROVED** [review]

**Pull request merged**